File: Podstawy Programowania/L3/3.c
//Mateusz Slotwinski

#include <stdio.h>
#include <stdbool.h>
#include <math.h>

int main() {
	puts("Witamy w kalkulatorze");
	while (true) {
		
		puts("Wybierz dzialanie matematyczne");
		puts("1. Dodawanie");
		puts("2. Odejmowanie");
		puts("3. Mnozenie");
		puts("4. Reszta z dzielenia");
		puts("0. Zakoncz");
		int tryb;
		scanf("%d", &tryb);

		if (tryb == 0) break;
		if (tryb > 4) {
			puts("Nieprawidlowa operacja");
			continue;
		}
		
		double x_1;
		puts("Wybierz pierwsza liczbe");
		scanf("%lf", &x_1);
		
		double x_2;
		puts("Wybierz druga liczbe");
		scanf("%lf", &x_2);
		
		switch (tryb) {
			case 1: 
				printf("%.2lf + %.2lf = %.2lf\n", x_1, x_2, x_1 + x_2);
				break;
			case 2: 
				printf("%.2lf - %.2lf = %.2lf\n", x_1, x_2, x_1 - x_2);
				break;
			case 3: 
				printf("%.2lf * %.2lf = %.2lf\n", x_1, x_2, x_1 * x_2);
				break;
			case 4: 
				while (true) {
					if (roundf(x_1) != x_1 || roundf(x_2) != x_2) {
						puts("Nieprawidlowe dane - oba argumenty musza byc calkowite");
					}
					
					if (roundf(x_1) != x_1) {
						puts("Wybierz pierwsza liczbe");
						scanf("%lf", &x_1);
					}
					
					if (roundf(x_2) != x_2) {
						puts("Wybierz druga liczbe");
						scanf("%lf", &x_2);
					}
					
					if (roundf(x_1) == x_1 && roundf(x_2) == x_2) {
						printf("%d %% %d = %d \n", (int)x_1, (int)x_2, (int)x_1 % (int)x_2);
						break;
					};
				}
				
				break;
		}
	}
	return 0;
}

File: Podstawy Programowania/Go/L3/3.go
package main

import "fmt"

func main() {
	fmt.Println("Witamy w kalkulatorze")
	for true {

		fmt.Println("Wybierz dzialanie matematyczne")
		fmt.Println("1. Dodawanie")
		fmt.Println("2. Odejmowanie")
		fmt.Println("3. Mnozenie")
		fmt.Println("4. Reszta z dzielenia")
		fmt.Println("0. Zakoncz")

		var tryb int
		fmt.Scan(&tryb)

		if tryb == 0 {
			break
		}
		if tryb > 4 {
			fmt.Println("Nieprawidlowa operacja")
			continue
		}

		var x_1 float64
		fmt.Println("Wybierz pierwsza liczbe")
		fmt.Scan(&x_1)

		var x_2 float64
		fmt.Println("Wybierz druga liczbe")
		fmt.Scan(&x_2)

		switch tryb {
		case 1:
			fmt.Printf("%.2f + %.2f = %.2f\n", x_1, x_2, x_1+x_2)
		case 2:
			fmt.Printf("%.2f - %.2f = %.2f\n", x_1, x_2, x_1-x_2)
		case 3:
			fmt.Printf("%.2f * %.2f = %.2f\n", x_1, x_2, x_1*x_2)
		case 4:
			for true {
				if float64(int(x_1)) != x_1 || float64(int(x_2)) != x_2 {
					fmt.Printf("Nieprawidlowe dane - oba argumenty musza byc calkowite")
				}

				if float64(int(x_1)) != x_1 {
					fmt.Printf("Wybierz pierwsza liczbe")
					fmt.Scan("%lf", &x_1)
				}

				if float64(int(x_2)) != x_2 {
					fmt.Printf("Wybierz druga liczbe")
					fmt.Scan("%lf", &x_2)
				}

				if float64(int(x_1)) == x_1 && float64(int(x_2)) == x_2 {
					fmt.Printf("%d %% %d = %d \n", int(x_1), int(x_2), int(x_1)%int(x_2))
					break
				}
			}
		}

	}
}
